server/query.py

```diff
@@ -229,6 +229,27 @@ def resultFunc():
     return resultFunc
 
 
+def csv_safe_unicode(row, ignoreErrors=True):
+    """
+    Given an array of values, make sure all strings are unicode in Python 3 and
+    str in Python 2.
+
+    :param row: an array which could contain mixed types.
+    :param ignoreErrors: if True, convert what is possible and ignore errors.
+        If false, conversion errors will raise an exception.
+    :returns: either the original array if safe, or an array with all byte
+        strings converted to unicode in Python 3 or str in Python 2.
+    """
+    # This is unicode in Python 2 and bytes in Python 3
+    not_str = six.text_type if str == six.binary_type else six.binary_type
+    if not any([isinstance(value, not_str) for value in row]):
```

Comment:
[Let's change this conditional for readability.](https://derickbailey.com/2014/03/27/double-negatives-in-code-dont-not-avoid-double-negative-boolean-logic/)

```
is_str = six.binary_type if str == six.binary_type else six.text_type
if all([isinstance(value, is_str) for value in row]):
```

Reply:
Since values can be types other than text or bytes, this isn't quite the same.  We want to leave other types (like integers) alone.

Reply:
Good point. In any case, let's remove the `[` `]` since I think that will cause the traversal to happen twice.